src/core/core.layouts.js

```diff
@@ -49,9 +99,152 @@ function findMaxPadding(boxes) {
 }
 
 function addSizeByPosition(boxes, size) {
-	helpers.each(boxes, function(box) {
+	var i, ilen, box;
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		box = boxes[i].box;
 		size[box.position] += box.isHorizontal() ? box.height : box.width;
-	});
+	}
+}
+
+function getMinSize(layout, params) {
+	var box = layout.box;
+	var minSize;
+
+	if (layout.isHorizontal) {
+		minSize = box.update(box.fullWidth ? params.width : params.chartAreaWidth, params.hBoxMaxHeight);
+		params.chartAreaHeight -= minSize.height;
+	} else {
+		minSize = box.update(params.vBoxMaxWidth, params.chartAreaHeight);
+		params.chartAreaWidth -= minSize.width;
```

Comment:
I wouldn't expect a method named `getMinSize` to update the `chartAreaWidth`/`chartAreaHeight`. Perhaps we could just return `minSize` here and then in `updateMinSizes` do:

```
layout.minSize = getMinSize(layout, params);
if (layout.isHorizontal) {
	params.chartAreaHeight -= minSize.height;
} else {
	params.chartAreaWidth -= minSize.width;
}
```